rust/lance-encoding/src/encodings/logical/primitive.rs

```diff
@@ -3346,7 +3346,15 @@ impl PrimitiveFieldEncoder {
                 row_number: 0, // legacy encoders do not use
             })
         })
-        .map(|res_res| res_res.unwrap())
```

Comment:
This is probably less relevant with this PR, but previously the `unwrap` here causing the program to crash since the encoding task error is not handled